### ISSUE TITLE
Request errors never get set on form

### ISSUE THREAD

**Issue by ncphillips** [laravel/precognition#82]
### Laravel Precognition Plugin Version

0.5.6

### Laravel Version

10.48.10

### Plugin

Vue w/ Inertia

### Description

Validation requests are sent to the server, which responds with the appropriate errors and HTTP status 422.

The errors are visible in the network tab, but they are never set on the Form object itself.

### Steps To Reproduce

I'm trying to setup precognition with Vue + Inertia. 

The request is being made and the errors are being returned, but it's not populating the form's errors object. It remains as an empty object.

web.php
```php
Route::name('store')
    ->post('/', [CourseAssignmentController::class, 'store'])
    ->middleware([HandlePrecognitiveRequests::class]);
```

Controller Method Signature
```php
public function store(CreateAssignmentRequest $request)
```

CreateAssignmentRequest
```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

class CreateAssignmentRequest extends FormRequest
{
    public function rules(): array
    {
        return [
            'title' => 'required|string',
            'overview' => 'string|nullable',
        ];
    }
}
```

Vue Component
```ts
import {useForm} from 'laravel-precognition-vue-inertia';

const form = useForm('post', route('courses.assignment.store', {course: props.course.id}), {
  title: '',
  overview: '',
});

function createAssignment() {
  form.submit({
    preserveScroll: true,
    onSuccess: () => form.reset(),
  });
}
```

**Comment by ncphillips** [comment]
What kind of more info would need @driesvints, @timacdonald ? I could try to get a test app to reproduce, send a loom, or hop on a call.

**Comment by timacdonald** [comment]
It would be good to see the entire Vue file and a reproduction repository would be even better. I'm unable to reproduce the issue with what I can see here.

**Comment by driesvints** [comment]
@ncphillips try reproducing the issue on a fresh repo so Tim can have a better look. Thanks!

```bash
laravel new bug-report --github="--public"
```

**Comment by ncphillips** [comment]
Interesting! I'll see what I can do. 

I feel our setup must have some assumed thing setup incorrectly

**Comment by timacdonald** [comment]
Thanks, @ncphillips. Keen to help out where we can to get this sorted for ya.

Let us know how you go.

**Comment by ncphillips** [comment]
I haven't had any luck :( I spun a new app up and everything worked fine.

I went back and made sure to upgrade all the inertia, precognition, axios, etc.

I feel like the most likely culprit is our version of laravel ? Really not sure.

**Comment by timacdonald** [comment]
I don't believe there has been any changes to the framework's precognition code since `10.11.0`.

Could you provide the full vue file you are having issues with?

Do you have any global configuration that could be impacting this?

**Comment by timacdonald** [comment]
@ncphillips, I'm gonna close this one. I'm certainly up for helping get to the bottom of it. If you can share any more information we can keep digging in.

**Comment by ncphillips** [comment]
That's fair @timacdonald, thanks!

**Comment by ncphillips** [comment]
Okay I just tried again from scratch.

Here's the JS package version I got
```
"laravel-precognition-vue-inertia": "^0.5.6",
```

I just whipped up a brand new page with this Vue template:
```vue
<template>
  <form @submit="submit">
    <label>
      Name:
      <input id="name" label="Name" v-model="form.name" @change="form.validate('name')" />
      <span v-if="form.errors.name">{{ form.errors.name }}</span>
    </label>
    <button type="submit" :disabled="form.processing">Submit</button>
    <div v-if="form.validating">Validating</div>
    <div v-if="form.submitting">Submitting</div>
    <div>
      <pre><data>{{ JSON.stringify(form.errors) }}</data></pre>
    </div>
  </form>
</template>
<script setup lang="ts">
import {useForm} from 'laravel-precognition-vue-inertia';

const form = useForm('post', '/banana', {
  name: '',
});

const submit = () =>
  form.submit({
    preserveScroll: true,
    onSuccess: () => form.reset(),
  });
</script>
```

These are the routes in my `web.php`:

```php
Route::get('banana', function () {
    return inertia('Banana');
});


Route::post('banana', function (\App\Http\Requests\BananaRequest $request) {
    return redirect()->back()->with('success', 'Hooray!');
})->middleware([HandlePrecognitiveRequests::class]);
```

Here is the Request 

```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

class BananaRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     */
    public function authorize(): bool
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
     */
    public function rules(): array
    {
        return [
            'name' => 'required|min:3'
        ];
    }
}
```

I do notice that it installed a different version of axios
<img width="1106" alt="image" src="https://github.com/user-attachments/assets/d66d9ffe-8f10-4478-b699-e2918824ac9f">

But once again, I see that it is returning with a 422 and the error messags, but those error messags are not in my form

<img width="1320" alt="image" src="https://github.com/user-attachments/assets/a1fc596a-e0ae-460d-b3fb-a8785633f9b1">

**Comment by timacdonald** [comment]
I copied this all verbatim into a fresh Laravel breeze application using Laravel `10.48.10` and Precognition `0.5.6` and everything seems to be working as expected.

https://github.com/user-attachments/assets/5f9b96c1-d57b-46c5-9282-8bfec7830580

Couple more things I'd love to know:

- Could you please share the response headers?
- Are there any messages in the JS console?
- Do you have any special CORS configuration?

I'm also wondering if you are doing any global configuration in this application that could be impacting this somehow?

**Comment by ncphillips** [comment]
> **I copied this all verbatim into a fresh Laravel breeze application using Laravel 10.48.10 and Precognition 0.5.6 and everything seems to be working as expected.**

Me too, it's really weird. I absolutely think it's something to do with our setup. This app's been around a while and has some unconventional things going on in various places.

> **Could you please share the response headers?**
<img width="752" alt="image" src="https://github.com/user-attachments/assets/ce790a1e-c114-436c-b4a1-9782303cc725">

> **Are there any messages in the JS console?**

I'm seeing some cookie warnings. _However,_ I also see them when I submit the form, which works just fine.

<img width="1318" alt="image" src="https://github.com/user-attachments/assets/e8e5677f-bbbd-408b-8e48-3651f16e0b8e">


> **Do you have any special CORS configuration?**

I am not aware of any special CORS config. Also, I do seem to be getting a valid response? Unless 422 is not what precognition expects.

> **I'm also wondering if you are doing any global configuration in this application that could be impacting this somehow?**

Maybe. But I can't think of what it might be.

**Comment by ncphillips** [comment]
Progress!

<img width="1281" alt="image" src="https://github.com/user-attachments/assets/c0579e36-e7ae-42c4-b213-1dfa5551c100">

Our errors are wrappe din another layer of `data`.

I suspect we have a middleware that wraps all json responses in `data`

**Comment by ncphillips** [comment]
**BOOM**

<img width="529" alt="image" src="https://github.com/user-attachments/assets/a2fa7b01-e039-43b0-909f-1fff8561dc14">

We are in the process of moving form a SPA to Inertia. For the SPA we have a custom ExceptionHandler that wraps any thrown exceptions into a palatable data structure. 

I just needed to throw a little `isPrecognition()` check in that thing to get it working!

<img width="840" alt="image" src="https://github.com/user-attachments/assets/8cd2f779-11ec-46ef-a106-e00773e1b457">

**Comment by timacdonald** [comment]
This is great! Glad to hear you were able to get this working and thanks for the update.

Hope you enjoy the package!

**Comment by ncphillips** [comment]
Looks great so far! I think my only complaint is `useForm` returns `any`. Might look at contributing an improvement there
